storage/txcache/concurrentMap.go

```diff
@@ -0,0 +1,134 @@
+package txcache
+
+import (
+	"sync"
+)
+
+// ConcurrentMap is a "thread" safe map of type string:Anything.
+// To avoid lock bottlenecks this map is dived to several (ShardCount) map shards.
+type ConcurrentMap struct {
+	shardCount int
```

Comment:
i think we might use "chunks" here, as it is pretty confusing with shards from the architecture. only a suggestion.